### ISSUE TITLE
Pipe text (or email) to oysttyer /dm

### ISSUE THREAD

**Issue by LwsBtlr** [oysttyer/oysttyer#108]
I was hoping I wold be able to use oysttyer to pipe the output of alert emails from my server into it to have them delivered as DMs. If this is possible, I haven't figured it out. If not, it would be a great addition.

Edit: I just found some old documentation for TTYter and see that I can do this, and it still works in oysttyer. I didn't get a man page when I installed oysttyer from ports, so sorry. (Leaving this in case someone else is searching for it.

`oysttyer -oathkey=<key> -oathsecret=<secret> -status /dm @<users> $TEXT`

**Comment by funnelfiasco** [comment]
You should also be able to pipe. Both of the commands below worked for me:

```echo "d funnelfiasco testing" | oysttyer
echo "/dm funnelfiasco testing" | oysttyer
```

One thing to be aware of is that if you have `verify` set, piping will honor that setting (thus not posting the tweet/DM).

I keep meaning to work on the documentation more, it just never happens. We don't have a man page, but once the online documentation is done, that's a possibility.

**Comment by LwsBtlr** [comment]
Thanks, that's a much better syntax for what I want to do.

One thing I noticed and I don't want to open a new "issue" is that the dm is restricted to 140 characters, even though I can send much longer DMs via the web or a GUI client. (not sure how long, but over 1000 characters for sure).

**Comment by funnelfiasco** [comment]
Is that only when piping or do you get that when using it interactively? We've supported longer DMs for a while. Can you check your version with `/vcheck`?

**Comment by classilla** [comment]
> You should also be able to pipe. Both of the commands below worked for me:
> 
> ```echo "d funnelfiasco testing" | oysttyer
> echo "/dm funnelfiasco testing" | oysttyer
> ```

`-runcommand` would probably be a little more helpful here since it wouldn't
clash with settings like `verify` (e.g.,

```
oysttyer -runcommand="/dm yomama sofat howfatisshe"
```

**Comment by LwsBtlr** [comment]
I installed the script today. ` -- your version of oysttyer is up to date (2.8.1).`

I tried to pipe in a string of 230 characters and it failed.

```
oysttyer -oauthkey=<key> -oauthsecret=<secret> -status="/dm @<me> 12345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890"
-- using SSL for default URLs.
trying to find cURL ... /usr/local/bin/curl
sorry, status too long: reduce by 102 chars, or use -autosplit={word,char,cut}.
```

EDIT: It works with -runcommand.

**Comment by funnelfiasco** [comment]
It also works with stdin. I spun off a new issue (109) for it not working with `-status`, but as classzilla noted that's probably not the best way to do this anyway.

**Comment by classilla** [comment]
> I installed the script today. -- your version of oysttyer is up to date (2.8.1). I tried to pie in a string of 230 characters and it failed.
> 
> `oysttyer -oauthkey=<key> -oauthsecret=<secret> -status="/dm @<me> 12345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890"

That will post a message "/dm @me ..." which I don't think you want to do.
You mean -runcommand here.
